### PR TITLE
Move integration test runner code to devtools_shared

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This moves some of the integration test runner code to `package:devtools_shared` so it can be re-used by `package:devtools_extensions`.

I will go through and comment on the parts of this PR that are copy / paste changes.

### CODE REVIEW

File: packages/devtools_shared/lib/src/test/integration_test_runner.dart [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+          // Debug outputs from the test will not show up in profile mode. Since
+          // we rely on debug outputs for detecting errors and exceptions from the
+          // test, we cannot run this these tests in profile mode until this issue
+          // is resolved.  See https://github.com/flutter/flutter/issues/69070.
+          // '--profile',
+          '--driver=$testDriver',
+          '--target=$testTarget',
+          '-d',
+          headless ? 'web-server' : 'chrome',
+          for (final arg in dartDefineArgs) '--dart-define=$arg',
```
Comment (kenzieschmoll) [inline packages/devtools_shared/lib/src/test/integration_test_runner.dart:51]: The only change to this class is to how these arguments are added and the parameter list for this method changed slightly. This used to be part of run_test.dart above.

File: packages/devtools_shared/lib/src/test/integration_test_runner.dart [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+          throw Exception(exceptionBuffer.toString());
+        }
+      }
+    }
+
+    await runTest(attemptNumber: 0);
+  }
+}
+
+class _IntegrationTestResult {
```
Comment (kenzieschmoll) [inline packages/devtools_shared/lib/src/test/integration_test_runner.dart:146]: rename is the only change to this class. Moved from run_test.dart

File: packages/devtools_shared/lib/src/test/integration_test_runner.dart [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+  String toString() {
+    if (result) {
+      return 'Test passed';
+    }
+    return 'Test \'$methodName\' failed: $details.';
+  }
+}
+
+class IntegrationTestRunnerArgs {
+  IntegrationTestRunnerArgs(
```
Comment (kenzieschmoll) [inline packages/devtools_shared/lib/src/test/integration_test_runner.dart:181]: base class created from the original TestRunnerArgs class in run_tests.dart. Stripped away any args that were specific to devtools app for this base class

File: packages/devtools_shared/lib/src/test/integration_test_runner.dart [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+        valueHelp: '1/3',
+        help: 'The shard number for this run out of the total number of shards '
+            '(e.g. 1/3)',
+      );
+    addExtraArgs?.call(argParser);
+    return argParser;
+  }
+}
+
+Future<void> runOneOrManyTests<T extends IntegrationTestRunnerArgs>({
```
Comment (kenzieschmoll) [inline packages/devtools_shared/lib/src/test/integration_test_runner.dart:277]: generalized helper method created from original code in run_tests.dart

File: packages/devtools_shared/lib/src/test/io_utils.dart [hunk 5]
```diff
@@ -8,7 +8,21 @@ import 'dart:async';
...
+}
+
+/// Return the user's home directory.
+String getUserHomeDir() {
+  final String envKey =
+      Platform.operatingSystem == 'windows' ? 'APPDATA' : 'HOME';
+  final String? value = Platform.environment[envKey];
+  return value == null ? '.' : value;
+}
+
```
Comment (kenzieschmoll) [inline packages/devtools_shared/lib/src/test/io_utils.dart:25]: copy pasted from utils_io

**Review by polina-c — APPROVED** [review]

**Pull request merged**